File: airbyte-integrations/connectors/destination-e2e-test/src/test-integration/java/io/airbyte/integrations/destination/e2e_test/TestingSilentDestinationAcceptanceTest.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.e2e_test;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.fasterxml.jackson.databind.JsonNode;
import io.airbyte.cdk.integrations.standardtest.destination.DestinationAcceptanceTest;
import io.airbyte.commons.json.Jsons;
import io.airbyte.integrations.destination.e2e_test.TestingDestinations.TestDestinationType;
import io.airbyte.protocol.models.v0.AirbyteMessage;
import io.airbyte.protocol.models.v0.AirbyteRecordMessage;
import java.util.Collections;
import java.util.HashSet;
import java.util.List;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

public class TestingSilentDestinationAcceptanceTest extends DestinationAcceptanceTest {

  @Override
  protected String getImageName() {
    return "airbyte/destination-e2e-test:dev";
  }

  @Override
  protected JsonNode getConfig() {
    return Jsons.jsonNode(
        Collections.singletonMap("test_destination", Collections.singletonMap("test_destination_type", TestDestinationType.SILENT.name())));
  }

  @Override
  protected JsonNode getFailCheckConfig() {
    return Jsons.jsonNode(Collections.singletonMap("test_destination", Collections.singletonMap("test_destination_type", "invalid")));
  }

  @Override
  protected List<JsonNode> retrieveRecords(final TestDestinationEnv testEnv,
                                           final String streamName,
                                           final String namespace,
                                           final JsonNode streamSchema) {
    return Collections.emptyList();
  }

  @Override
  protected void setup(final TestDestinationEnv testEnv, final HashSet<String> TEST_SCHEMAS) {
    // do nothing
  }

  @Override
  protected void tearDown(final TestDestinationEnv testEnv) {
    // do nothing
  }

  @Override
  protected void assertSameMessages(final List<AirbyteMessage> expected,
                                    final List<AirbyteRecordMessage> actual,
                                    final boolean pruneAirbyteInternalFields) {
    assertEquals(0, actual.size());
  }

  @Override
  // Skip because `retrieveRecords` returns an empty list at all times.
  @Disabled
  @Test
  public void testSyncNotFailsWithNewFields() {}

  @Override
  // This test assumes that dedup support means normalization support.
  // Override it to do nothing.
  @Disabled
  @Test
  public void testIncrementalDedupeSync() throws Exception {
    super.testIncrementalDedupeSync();
  }

}

File: airbyte-integrations/connectors/destination-dev-null/src/test-integration-legacy/kotlin/io/airbyte/integrations/destination/dev_null/TestingSilentDestinationAcceptanceTest.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.integrations.destination.dev_null

import com.fasterxml.jackson.databind.JsonNode
import io.airbyte.cdk.integrations.standardtest.destination.DestinationAcceptanceTest
import io.airbyte.cdk.util.Jsons
import io.airbyte.protocol.models.v0.*
import java.util.*
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test

open class TestingSilentDestinationAcceptanceTest : DestinationAcceptanceTest() {
    override val imageName = "airbyte/destination-dev-null:dev"
    override val isCloudTest = false

    override fun getConfig(): JsonNode {
        return Jsons.readTree("""{"test_destination": {"test_destination_type": "SILENT"}}""")
    }

    override fun getFailCheckConfig(): JsonNode {
        return Jsons.readTree("""{"test_destination": {"test_destination_type": "invalid"}}""")
    }

    override fun retrieveRecords(
        testEnv: TestDestinationEnv?,
        streamName: String,
        namespace: String,
        streamSchema: JsonNode
    ): List<JsonNode> {
        return emptyList()
    }

    override fun setup(testEnv: TestDestinationEnv, TEST_SCHEMAS: HashSet<String>) {
        // do nothing
    }

    override fun tearDown(testEnv: TestDestinationEnv) {
        // do nothing
    }

    override fun assertSameMessages(
        expected: List<AirbyteMessage>,
        actual: List<AirbyteRecordMessage>,
        pruneAirbyteInternalFields: Boolean
    ) {
        Assertions.assertEquals(0, actual.size)
    }

    open override fun getDefaultSchema(config: JsonNode): String? {
        return super.getDefaultSchema(config) ?: "default_schema"
    }

    // Skip because `retrieveRecords` returns an empty list at all times.
    @Disabled @Test override fun testSyncNotFailsWithNewFields() {}

    @Disabled @Test override fun testAirbyteTimeTypes() {}

    // This test assumes that dedup support means normalization support.
    // Override it to do nothing.
    @Disabled
    @Test
    @Throws(Exception::class)
    override fun testIncrementalDedupeSync() {
        super.testIncrementalDedupeSync()
    }
}
